### DISCUSSION TITLE
X-frame Options to embed pdfs?

### DISCUSSION THREAD

**Discussion by Jeet59** [pocketbase/pocketbase#3117] [Q&A]
Will this enable us to upload pdfs and then embed them in our frontend??I am using React for my frontend and having a nightmare trying to get the pdfs working. Classmate suggested PocketBase so any kind of input on the X-Frame Options headers for the uploaded data might be useful.

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand what do you mean, but uploaded files are not served with `X-Frame-Options` header.

As far as I know this header is to explicitly control where and which file resource can be loaded and previewed (_aka. similar to the [frame-ancestor CSP directive](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-ancestors)_) and as long as the pdf is served with `Content-Disposition: inline` (which PocketBase does) you should be able to embed and preview the PDF just fine (_if the browser supports it_).

**Reply by Jeet59** [reply]
Previously I was using s3 with cloudfront and I was getting the X-Frame Headers error everytime I tried to load the file.

**Reply by ganigeorgiev** [reply]
The file serve endpoint (aka. `GET /api/files/:collectionId/:recordId/:filename`) doesn't add the `X-Frame-Options` header to the response.
We don't have control over the file response if you are loading the file directly from S3/CloudFront.

**Reply by ganigeorgiev** [reply]
_If you are serving files directly with CloudFront you could check [AWS ResponseHeadersPolicy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-responseheaderspolicy-frameoptions.html)_

**Reply by Jeet59** [reply]
I was rendering the pdf directly on the frontend fetching the cloudfront link directly. It adds the X-Frame Options header with SAMEORIGIN value. I have read the policy before which states only two values are allowed to the header- DENY or SAMEORIGIN. I might be wrong but the value should be ALLOWALL if we want to embed the file. I have no knowledge of the Content-Disposition header so I'll try working out with PocketBase hopefully that does the work.

**Reply by ganigeorgiev** [reply]
I don't think `X-Frame-Options` has `ALLOWALL` (*_it seems that there is actually such value in the [spec](https://html.spec.whatwg.org/multipage/document-lifecycle.html#the-x-frame-options-header) but it is marked as legacy_). Remove the header if you don't want to apply any constraint to the file embedding.

The `Content-Disposition:inline/attachment` header is used to explicitly tell the browser whether to preview or download the served file.
